wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/gearsbot/subsystems/Drivetrain.java

```diff
@@ -131,6 +127,63 @@ public double getDistanceToObstacle() {
     return m_rangefinder.getAverageVoltage();
   }
 
+  /**
+   * Drive until the robot is the given distance away from the box. Uses a local PID controller to
+   * run a simple PID loop that is only enabled while this command is running. The input is the
+   * averaged values of the left and right encoders.
+   *
+   * @param distance The distance away from the box to drive to
+   * @return A command that drives the robot to the provided distance based on the rangefinder.
+   */
+  public Command driveDistanceFromObstacle(double distance) {
+    return driveDistance(-distance, () -> -getDistanceToObstacle());
```

Comment:
Why are these negated?

Reply:
They're negated so that the robot drives forward in order to reduce distance. It's a bit jank, but better than the previous version's negative P gain.